### PR TITLE
chore: updated test cases according to new flow for clevertap

### PR DESCRIPTION
**Pull request by ItsSudip**
## What are the changes introduced in this PR?

We are moving proxy test cases of clevertap to new format.

## What is the related Linear task?

Resolves INT-1479

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3123/test-report.html

**Review by utsabc — APPROVED** [review]

File: test/integrations/destinations/clevertap/dataDelivery/business.ts [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+            status: 401,
+            message: 'Request failed  with status: 401',
+            response: [
+              {
+                metadata: generateMetadata(123),
+                error: '{"status":"fail","error":"Invalid Credentials","code":401}',
+                statusCode: 401,
+              },
+            ],
+            statTags: {
```
Comment (yashasvibajpai) [inline test/integrations/destinations/clevertap/dataDelivery/business.ts:136]: Lets declare them (statTags) as variables?

File: test/integrations/destinations/clevertap/dataDelivery/business.ts [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+      response: {
+        status: 200,
+        body: {
+          output: {
+            status: 400,
+            message: 'Request failed  with status: 200',
+            response: [
+              {
+                metadata: generateMetadata(123),
+                error: '{"status":"fail","processed":0,"unprocessed":[]}',
```
Comment (yashasvibajpai) [inline test/integrations/destinations/clevertap/dataDelivery/business.ts:200]: Is it possible to add a partial failure response case too?
like 
`{"status":"fail","processed":1,"unprocessed": 3, reason[ { some error for some record}]}`

File: test/integrations/destinations/clevertap/dataDelivery/business.ts [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+                statusCode: 400,
+              },
+            ],
+            statTags: {
```
Comment (yashasvibajpai) [inline test/integrations/destinations/clevertap/dataDelivery/business.ts:204]: same, can be declared and reused

**Review by yashasvibajpai — APPROVED** [review]
General comments, approving

**Review by yashasvibajpai — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**